homeworks/week6/hw6.py

```diff
@@ -0,0 +1,49 @@
+from collections import Counter
+def getShortestSubstrOfAllChar(S, T):
+    '''get shortest substring of all char. If there are multiple substrings with same length, return the leftmost one.
+
+    Time Complexity: O(max(N, M)) where N = length of S and M = length of T (in case O(N) < O(M) the time complexity is strictly O(N), it does not depend on the size of the alphabet |A|).
```

Comment:
The code performs a linear iteration on both string; hence, it should be O(N+M).